app/src/infra/api.rs

```diff
@@ -186,9 +186,15 @@ impl HostService {
                 .context("Error clearing cookie")
         };
         let parse_token = move |data: String| {
-            get_claims_from_jwt(&data)
+            serde_json::from_str::<login::ServerLoginResponse>(&data)
                 .context("Could not parse response")
-                .and_then(set_cookies)
+                .and_then(|r| {
+                    let success = get_claims_from_jwt(r.token.as_str())
+                        .context("Could not parse response")
+                        .and_then(set_cookies)
+                        .is_ok();
```

Comment:
Hmm, sorry I'm reviewing on my phone so it's a bit hard for me to tell exactly what's going on here, but:
- you get a Result
- you add more info to the error with context()
- if it was no error, you set the cookies
- and then you *throw away the error* and just return whether it was successful (is_ok)

I'll have a closer look later, why this function returns a bool, but I'd like to avoid throwing away errors.

Reply:
Yeah, the problem is that it expects a result with a Result<(login::ServerLoginResponse, bool) and I'm not sure how I should change. I'm actually happy that I got it working. I find it really hard to read Rust code in general, sorry :(

Reply:
Okay, I checked out your branch to have a better look, and I see the issue. The String returned here is the username, not the token. You were right in parsing the server login response and _then_ calling `get_claims_from_jwt`, but you can just stop after `and_then(set_cookies)` like the original code. The callback still takes a String

Reply:
Oof, I overcomplicated it. Should be fixed now